src/main/java/org/jenkinsci/test/acceptance/po/Control.java

```diff
@@ -94,10 +95,39 @@ public void check(boolean state) {
         check(resolve(), state);
     }
 
+    /**
+     * sends a click on the underlying element.
+     * You should not use this on any page where the click will cause another page to be loaded as it will not
+     * gaurantee that the new page has been loaded.
+     * @see #clickAndWaitToBecomeStale()
+     * @see #clickAndWaitToBecomeStale(Duration)
+     */
     public void click() {
         resolve().click();
     }
 
+
+    /**
+     * like click but will block for up to 30 seconds until the underlying web element has become stale.
+     * see https://blog.codeship.com/get-selenium-to-wait-for-page-load/
+     */
+    public void clickAndWaitToBecomeStale() {
```

Comment:
I do not quite like this new API (majority of ATH should switch to). We will need to give this deeper look to identify a cleaner path forward.

Reply:
there will be some things where a click does not reload the page.  doing this in click would be bad as then we are not mirroring the selenium API in the page object as well as we would need a clickButDoNotWaitForWhatever for these.
Happy to hear your thoughts

Reply:
@olivergondza @raul-arabaolaza  What can be done to move this forward?  I don't see that adding this API has any significant downside, or that there is a better way to implement this.
cc: @jglick

Reply:
`clickAndWaitForNewPage()` may be clearer...
maybe make this `@Restricted(NoExternalUse.class)` would be good enough, to move forward and allow for later refactoring.
currently the methods map the Selenium ones so changing click to not be the same would surprise some users (even if it is suprising that click does not do what many people may expect...

Reply:
> there will be some things where a click does not reload the page

I love the idea of doing this in `click()` or so @olivergondza, but James' point is very true too: I do not think every clicks reload the pages, so we would also create issue by changing click()'s behavior to wait for the page to become stale by default too isn't it?

Reply:
Shouldn't this new method be named `submit()` ? As it would pretty much match the (expected) behaviour of webElement.submit(), which seem to have some restrictions (https://www.quora.com/What-is-the-difference-between-click-and-submit-in-webdriver).

Reply:
I think naming it `submit()` might cause the same sort of confusion as changing the behavior of `click()`.  

Why not create the new method as `click(int timeout)` - this would let existing code continue to work (or flake) as it does today, and we can selectively start creating PR's to fix the instances where we would want to actually use the new method?

See #523